Index: BaseObject.py
===================================================================
--- BaseObject.py	(revision 7383)
+++ BaseObject.py	(working copy)
@@ -596,8 +596,15 @@
         schemata = self.Schemata()
         fields = []
 
+        #raise 'fieldset is %s' % fieldset
+
         if fieldset is not None:
-            fields = schemata[fieldset].fields()
+            if type(fieldset) == type([]):
+                for fset in fieldset:
+                    fields += schemata[fset].fields()
+            else:
+                fields += schemata[fieldset].fields() 
+            #raise str(fields)
         else:
             if data: fields += schema.filterFields(isMetadata=0)
             if metadata: fields += schema.filterFields(isMetadata=1)
Index: Schema/__init__.py
===================================================================
--- Schema/__init__.py	(revision 7383)
+++ Schema/__init__.py	(working copy)
@@ -19,6 +19,8 @@
 from Globals import InitializeClass
 from Products.CMFCore import permissions
 
+
+
 __docformat__ = 'reStructuredText'
 _marker = []
 
@@ -545,8 +547,14 @@
 
         if fieldset is not None:
             schemata = instance.Schemata()
-            fields = [(field.getName(), field)
-                      for field in schemata[fieldset].fields()]
+            if type(fieldset) == type([]):
+                for fset in fieldset:
+                    fields = [(field.getName(), field)
+                          for field in schemata[fset].fields()]
+            else:
+                fields = [(field.getName(), field)
+                          for field in schemata[fieldset].fields()]
+            
         else:
             if data:
                 fields.extend([(field.getName(), field)
@@ -562,7 +570,9 @@
         for name, field in fields:
             error = 0
             value = None
+            
             widget = field.widget
+
             if form:
                 result = widget.process_form(instance, field, form,
                                              empty_marker=_marker)
